kura/org.eclipse.kura.container.provider/src/main/java/org/eclipse/kura/container/provider/ContainerInstance.java

```diff
@@ -120,6 +153,50 @@ public void onDisabled() {
         updateState(State::onDisabled);
     }
 
+    private ValidationResult validateContainerImageSignature(ContainerInstanceOptions configuration) {
+
+        if (Objects.isNull(this.availableContainerSignatureValidationService)
+                || this.availableContainerSignatureValidationService.isEmpty()) {
+            logger.warn("No container signature validation service available. Signature validation failed.");
+            return FAILED_VALIDATION;
+        }
+
+        Optional<String> optTrustAnchor = configuration.getSignatureTrustAnchor();
+        if (!optTrustAnchor.isPresent() || optTrustAnchor.get().isEmpty()) {
+            logger.warn("No trust anchor available. Signature validation failed.");
+            return FAILED_VALIDATION;
+        }
+
+        String trustAnchor = optTrustAnchor.get();
+        boolean verifyInTransparencyLog = configuration.getSignatureVerifyTransparencyLog();
+        Optional<RegistryCredentials> registryCredentials = configuration.getRegistryCredentials();
+
+        for (ContainerSignatureValidationService validationService : this.availableContainerSignatureValidationService) {
+            ValidationResult results = FAILED_VALIDATION;
+
+            try {
+                if (registryCredentials.isPresent()) {
+                    results = validationService.verify(configuration.getContainerImage(),
+                            configuration.getContainerImageTag(), trustAnchor, verifyInTransparencyLog,
+                            registryCredentials.get());
+                } else {
+                    results = validationService.verify(configuration.getContainerImage(),
+                            configuration.getContainerImageTag(), trustAnchor, verifyInTransparencyLog);
+                }
+            } catch (KuraException e) {
+                logger.warn(
```

Comment:
This should be error log level in my opinion

Reply:
I'm usually referencing this

![image](https://github.com/eclipse/kura/assets/22748355/c1eed2d2-1734-4b44-afd1-dd7304804c89)

Since the process can continue even though the process state is unwanted this is a "warning" level.